ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1.admin;
+
+option java_outer_classname = "MeteringReportOuterClass";
+option java_package = "com.daml.ledger.api.v1.admin";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1.Admin";
+
+import "google/protobuf/timestamp.proto";
+
+// Experimental API to extract metering reports.
+//
+// Authorization rules for the services RPCs are specified on the `RpcNameRequest`
+// messages as boolean expressions over these facts:
+// - ``HasRight(r)``: true iff the authenticated user has right `r`
+//
+service MeteringReportService {
+
+    // Run a report.
+    rpc Report (ReportRequest) returns (ReportResponse);
+
+}
+
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message ReportRequest {
+
+    // The from timestamp (inclusive)
+    google.protobuf.Timestamp from = 1;
+
+    // The to timestamp (exclusive)
+    google.protobuf.Timestamp to = 2;
+
+    // If provided only return results for the given applications
+    repeated string applications = 3;
+}
+
+message ApplicationReport {
+
+    // The application Id
+    string application = 1;
+
+    // The event count for the application
+    int64 events = 2;
+}
+
+message ReportResponse {
+
+    // The reporting participant
+    string participant = 1;
+
+    // The requested, and actual, from-time
+    google.protobuf.Timestamp from = 2;
```

Comment:
why do we use times here instead of offsets for what seems effectively a range on transactions where we usually use offsets? I get the ux aspect but that also applies to other things like pruning so having this one be the odd one out might not be great and we want instead a more general mechanism for translating times to offsets.

Reply:
Using offsets would not really work on Canton with multiple domains would it?

Reply:
I don’t know enough about multi-domain canton but if offsets work for pruning & transaction streams for multi-domain canton, I don’t understand why it doesn’t work for metering.

Reply:
In multi-domain canton the record time stamps are not guaranteed to be in-order; and the current integration is a bit hacky around that.

Event metering thus aims to provide an easy UX by aggregating events using the participant node's wall clock. We accept the limitation that duplicate participant nodes might be slighly out of sync wrt event-metering.